--- a/boost/thread/pthread/thread_data.hpp	2019-12-10 02:20:01.000000000 +0200
+++ b/boost/thread/pthread/thread_data.hpp	2022-10-16 20:31:36.011038033 +0200
@@ -57,6 +57,12 @@
 #else
           std::size_t page_size = ::sysconf( _SC_PAGESIZE);
 #endif
+
+//PATCH
+#undef PTHREAD_STACK_MIN
+#define PTHREAD_STACK_MIN 16384
+//PATCH
+
 #if PTHREAD_STACK_MIN > 0
           if (size<PTHREAD_STACK_MIN) size=PTHREAD_STACK_MIN;
 #endif
